### PR TITLE
Added integration test for verifying headers sent to the Platform.

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-1836" title="DX-1836" target="_blank">TaskDX-1836</a>  We have an integration test for platform request headers
  </td></table>
  <br />

### CODE REVIEW

**Comment by mitchell-as** [comment]
If this PR is accepted, it will need to be force-merged unless DX-2013 is solved before I'm back next week.

File: pkg/platform/api/api.go [hunk 1]
```diff
@@ -46,6 +47,13 @@ func (r *RoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 		return nil, platform.NewCountryBlockedError()
 	}
 
+	// This code block is for integration testing purposes only.
+	if os.Getenv(constants.PlatformApiRequestPrintHeadersEnvVarName) != "" &&
+		(condition.OnCI() || condition.BuiltOnDevMachine()) {
+		logging.Debug("User-Agent: %s\n", resp.Request.Header.Get("User-Agent"))
+		logging.Debug("X-Requestor: %s\n", resp.Request.Header.Get("X-Requestor"))
```
Comment (Naatan) [inline pkg/platform/api/api.go:54]: Please also add the URL (put it first).

File: internal/constants/constants.go [hunk 2]
```diff
@@ -494,3 +494,7 @@ const TerminalAnimationInterval = 150 * time.Millisecond
 // RuntimeSetupWaitEnvVarName is only used for an integration test to pause installation and wait
 // for Ctrl+C.
 const RuntimeSetupWaitEnvVarName = "ACTIVESTATE_CLI_RUNTIME_SETUP_WAIT"
+
+// PlatformApiRequestPrintHeadersSetEnvVarName is only used for an integration test to print some
+// expected headers from a Platform API request.
+const PlatformApiRequestPrintHeadersEnvVarName = "ACTIVESTATE_CLI_PLATFORM_API_REQUEST_PRINT_HEADERS"
```
Comment (Naatan) [inline internal/constants/constants.go:500]:
```suggestion
// PlatformApiRequestPrintRequestsEnvVarName is only used for an integration test to print some
// expected headers from a Platform API request.
const PlatformApiRequestPrintRequestsEnvVarName = "ACTIVESTATE_CLI_PLATFORM_API_REQUEST_PRINT_REQUESTS
```

**Review by Naatan — CHANGES_REQUESTED** [review]

**Review by Naatan — APPROVED** [review]

**Comment by mitchell-as** [comment]
@Naatan would you please force-merge this? The failing unit tests do not belong to this ticket and Target & Verify is still failing.

**Pull request merged**